media_kit_video_controls/lib/src/controls/methods/fullscreen.dart

```diff
@@ -0,0 +1,135 @@
+                filterQuality: state(context).widget.filterQuality,
+                controls: state(context).widget.controls,
+                // Do not acquire or modify existing wakelock in fullscreen mode:
+                wakelock: false,
+              ),
+            ),
+            transitionDuration: Duration.zero,
+            reverseTransitionDuration: Duration.zero,
+          ),
+        );
+        await enterNativeFullscreen();
+      }
+    }
+  });
+}
+
+/// Makes the [Video] present in the current [BuildContext] exit fullscreen.
+Future<void> exitFullscreen(BuildContext context) {
+  return lock.synchronized(() async {
+    if (isFullscreen(context)) {
+      if (context.mounted) {
+        Navigator.of(context).maybePop();
+        // It is known that this [context] will have a [FullscreenInheritedWidget] above it.
+        FullscreenInheritedWidget.of(context).parent.refreshView();
+      }
+      // [exitNativeFullscreen] is moved to [WillPopScope] in [FullscreenInheritedWidget].
+      // This is because [exitNativeFullscreen] needs to be called when the user presses the back button.
+      // await exitNativeFullscreen();
+    }
+  });
+}
+
+/// Toggles fullscreen for the [Video] present in the current [BuildContext].
+Future<void> toggleFullscreen(BuildContext context) {
+  if (isFullscreen(context)) {
+    return exitFullscreen(context);
+  } else {
+    return enterFullscreen(context);
+  }
+}
+
+/// Makes the native window enter fullscreen.
+Future<void> enterNativeFullscreen() async {
+  try {
+    if (kIsWeb) {
+      await document.documentElement?.requestFullscreen();
+    } else if (Platform.isAndroid || Platform.isIOS) {
+      await Future.wait(
+        [
+          SystemChrome.setEnabledSystemUIMode(SystemUiMode.immersiveSticky),
+          SystemChrome.setPreferredOrientations(
+            [
+              DeviceOrientation.landscapeLeft,
+              DeviceOrientation.landscapeRight,
+            ],
+          ),
+        ],
+      );
+    } else if (Platform.isMacOS || Platform.isWindows || Platform.isLinux) {
+      await const MethodChannel('com.alexmercerind/media_kit_video')
+          .invokeMethod(
+        'Utils.EnterNativeFullscreen',
+      );
+    }
+  } catch (exception, stacktrace) {
+    debugPrint(exception.toString());
+    debugPrint(stacktrace.toString());
+  }
+}
+
+/// Makes the native window exit fullscreen.
+Future<void> exitNativeFullscreen() async {
+  try {
+    if (kIsWeb) {
+      document.exitFullscreen();
+    } else if (Platform.isAndroid || Platform.isIOS) {
+      await Future.wait(
+        [
+          SystemChrome.setEnabledSystemUIMode(SystemUiMode.edgeToEdge),
+          SystemChrome.setPreferredOrientations([]),
+        ],
+      );
+    } else if (Platform.isMacOS || Platform.isWindows || Platform.isLinux) {
+      await const MethodChannel('com.alexmercerind/media_kit_video')
+          .invokeMethod(
+        'Utils.ExitNativeFullscreen',
```

Comment:
TODO:

Reply:
Why not use `window_manager`?

Reply:
Although [`window_manager`](https://pub.dev/packages/window_manager) is a good package, I'm (personally) kind of skeptical about it's stability.

A [number of fullscreen related issues](https://github.com/leanflutter/window_manager/issues?q=is%3Aissue+is%3Aopen+fullscreen) (more on Windows) are present in it. Although, Win32 (Windows) doesn't offer a simple "built-in" way to go fullscreen, I think GTK & macOS do (where a single method-call should work).

Probably it also changes the window-related attribute(s) & behavior at start-up, which may cause trouble to people who weren't using [`window_manager`](https://pub.dev/packages/window_manager).